pkg/pmem-device-manager/pmd-ndctl.go

```diff
@@ -46,6 +46,14 @@ func (pmem *pmemNdctl) GetCapacity() (map[string]uint64, error) {
 }
 
 func (pmem *pmemNdctl) CreateDevice(name string, size uint64, nsmode string) error {
+	// force-align namespace size up to next GB-boundary, to avoid alignement issues
+	var align uint64 = 1024 * 1024 * 1024
```

Comment:
I noticed Align option and tried to use it, but I did hit some other errors. It is possible something else was affecting operation then as it was relatively early in this change. Worth retrying now when other changes have cleared up.

Reply:
I re-tried setting opts.Align = 1GB in CreateNamespace, but this produces errors and Namespace can not be created, and seems to happen regardless of explicit size alignment by pmem-csi.
I tried this both on qemu-emulated and HW-based device. 
pmem-csi output is same on qemu and HW-based systems:
```
libndctl: ndctl_pfn_enable: pfn0.1: failed to enable
```
dmesg on qemu-based system:
```
nd_pmem pfn0.1: namespace0.2 unable to satisfy requested alignment
```

dmesg On HW-based device differs slightly:
```
[935035.760338] nd_pmem pfn1.3: namespace1.7 alignment collision, truncate 218103808 bytes
[935035.760342] nd_pmem pfn1.3: namespace1.7 unable to satisfy requested alignment
```

Based on these errors, I propose we keep pmem-csi level alignment setting and avoid use of opts.Align.
Also, we could ask on libndctl side is this expected what we see

Reply:
more alignment and sizes notes:
if I ask ndctl to create size=1G align=1G then libndctl just fails to create.
size=2G align=1G will create namespace, but resulting size is 1G as libndctl force-aligns by changing size down, which is not suitable for pmem-csi as we can not provision volume smaller than was asked.
Means, in pmem-csi we have to be prepared before starting creation:
- use libndctl option: align=1G
- size we ask is always at least 2GB
- resulting size cant become smaller-than-ask by libndctl force-aligning, this is achievable by force-align size up to next 1GB before creation.
That's even more checking and forcing as in current proposal (min. size to be added).
One idea, ask ndctl side does it make sense to create smaller-than-asked namespaces if there is space available to force-align up.

Reply:
I created [ndctl issue](https://github.com/pmem/ndctl/issues/79) with proposal to force-align up, but in pmem-csi side we have to implement above list of pre-checks for relaible operation

Reply:
@okartau so, till the issue fixed in libndctl we live with this work around.

Reply:
> @okartau so, till the issue fixed in libndctl we live with this work around. or we can move this check into ndctl wrapper code?

Reply:
we need this workaround and even more, as currently for small-size request we create sz=1G/align=1G which will fail. We also need to set minimum size to 2G to succeed.
Where these adjustments should be made, yes we can select proper place.

Reply:
actually, it appears after current PR code, we have force-alignment code both on device-manager level and in ndctl wrapper code, I try to unify so that only one remains (ndctl wrapper)